compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -1087,4 +1087,201 @@ mod tests {
+                String::from_utf8_lossy(&base64_decoded)
+            },
+        };
+
+        vec![
+            // Token::Ident(_)
+            (None, vec![
+                format!("
+                    let \"{}\" = ();
+                ", s),
+            ]),
+
+            (Some(Token::Str("".to_string())), vec![
+                // Token::Str(String)
+                format!("
+                    let s = \"{}\";
+                ", s),
+            ]),
+
+            (Some(Token::RawStr("".to_string(), 0)), vec![
+
+                // let s = r"Hello world";
+                format!("
+                    let s = r\"{}\";
+                ", s),
+
+                // let s = r#"Simon says "hello world""#;
+                format!("
+                    let s = r#\"{}\"#;
+                ", s),
+
+                // // Any number of hashes may be used (>= 1) as long as the string also terminates with the same number of hashes
+                // let s = r#####"One "#, Two "##, Three "###, Four "####, Five will end the string."#####; 
+                format!("
+                    let s = r##\"{}\"##;
+                ", s),
+                format!("
+                    let s = r###\"{}\"###;
+                ", s),
+                format!("
+                    let s = r####\"{}\"####;
+                ", s),
+                format!("
+                    let s = r#####\"{}\"#####;
+                ", s),
+            ]),
+
+            (Some(Token::FmtStr("".to_string())), vec![
+
+                format!("
+                    assert(x == y, f\"{}\");
+                ", s),
+            ]),
+
+            // expected token not found
+            //
+            // (Some(Token::LineComment("".to_string(), None)), vec![
+            //     format!("
+            //         //{}
+            //     ", s),
+            //
+            //     format!("
+            //         // {}
+            //     ", s),
+            // ]),
+
+            // expected token not found
+            //
+            // According to the docs: https://noir-lang.org/docs/noir/concepts/comments/
+            // "Start a block comment with /* and end the block with */."
+            (Some(Token::BlockComment("".to_string(), None)), vec![
+                // bug
+                // format!("
+                //     /*{}*/
+                // ", s),
+
+                // bug
+                // format!("
+                //     /* {} */
+                // ", s),
+
+                // format!("
+                //     /*\n{}\n*/
+                // ", s),
+
+            ]),
+        ]
+    }
+
+    #[test]
+    fn test_big_list_of_naughty_strings() {
+        use std::io::Cursor;
+        use std::io::Read;
+        use std::mem::discriminant;
+        use std::path::Path;
+        use std::process::Command;
+
+        let blns_url = "https://raw.githubusercontent.com/minimaxir/big-list-of-naughty-strings/master/blns.base64.json";
```

Comment:
Let's just vendor this. The file is small enough that the cost of adding it to git is small relative to the overhead from downloading each time / relative complexity of caching.

Reply:
Having looked at the list, I think only a subset of it is really relevant to us anyway.